dev/devicelab/bin/tasks/gradle_plugin_test.dart

```diff
@@ -208,11 +225,49 @@ Future<Null> _runGradleTask({String workingDirectory, String task}) async {
     throw 'Gradle exited with error';
 }
 
-Future<ProcessResult> _resultOfGradleTask({String workingDirectory, String task}) {
+Future<ProcessResult> _resultOfGradleTask({String workingDirectory, String task,
+    List<String> options}) {
+  final List<String> args = <String>['app:$task'];
+  if (options != null) {
+    args.addAll(options);
+  }
   return Process.run(
     './gradlew',
-    <String>['app:$task'],
+    args,
     workingDirectory: workingDirectory,
     environment: <String, String>{ 'JAVA_HOME': javaHome }
   );
 }
+
+class _Dependencies {
+  String target;
+  Set<String> dependencies;
+  _Dependencies(String depfilePath) {
+    final RegExp _separatorExpr = new RegExp(r'([^\\]) ');
+    final RegExp _escapeExpr = new RegExp(r'\\(.)');
+
+    // Depfile format:
+    // outfile1 outfile2 : file1.dart file2.dart file3.dart
+    final String contents = new File(depfilePath).readAsStringSync();
+    final List<String> colonSeparated = contents.split(': ');
+    target = colonSeparated[0].trim();
+    dependencies = colonSeparated[1]
+        .replaceAllMapped(
+        _separatorExpr, (Match match) => '${match.group(1)}\n')
+        .split('\n')
+        .map((String path) =>
+        path.replaceAllMapped(_escapeExpr, (Match match) => match.group(1))
+            .trim())
```

Comment:
Again: whitespace could make it easier to see that lines 259-260 are both part of the `map` on lines 258.